kernel/aster-nix/src/fs/path/dentry.rs

```diff
@@ -48,13 +49,14 @@ impl Dentry_ {
     fn new(inode: Arc<dyn Inode>, options: DentryOptions) -> Arc<Self> {
         Arc::new_cyclic(|weak_self| Self {
             inode,
-            flags: AtomicU32::new(DentryFlags::empty().bits()),
             name_and_parent: match options {
                 DentryOptions::Leaf(name_and_parent) => RwLock::new(Some(name_and_parent)),
                 _ => RwLock::new(None),
             },
             this: weak_self.clone(),
             children: Mutex::new(Children::new()),
+            mountpoint_count: AtomicU8::new(0),
+            flags: AtomicU32::new(DentryFlags::empty().bits()),
```

Comment:
Thank you for pointing this out, but `DentryFlags` may have other uses in the future besides marking mount points, as there are many flags defined in Linux.
[Linux dentry flags](https://elixir.bootlin.com/linux/v6.7/source/include/linux/dcache.h#L153)

Reply:
Currently, this flag seems redundant because the newly added `mountpoint_count` essentially represents the meaning of `DentryFlags::MOUNTED`. At least from now, I think there is no need to maintain an extra field, and `DentryFlags::MOUNTED` can be removed.